index.js

```diff
@@ -10,8 +10,13 @@
 var isUTF8 = require('is-utf8');
 var isBuffer = require('is-buffer');
 
+function maybeUTF8(buf) {
```

Comment:
I don't think this function is necessary. Just do the `isUTF8(buf.slice(0, 7))` in the `if` statement